### PR TITLE
Flavor support

### PR DESCRIPTION
**Pull request by kowczarz**
Co-authored with @mczernek & @staszekscp 

### Details
We wanted to enable developers and testers to install multiple app variant simultaneously on physical devices.

To enable it we used:
- On android: 4 different flavors with 2 build types, but since some of them are unused we have disabled them and final variants are presented in a table below 

flavor | release | debug
-- | -- | --
default | ✅ | ✅
AdHoc | ✅ | ❌
Dev | ✅ | ✅
e2e | ✅ | ❌

- on iOS: we used buildSchemes corresponding to variants from android.

### Fixed Issues
$ [#16805](https://github.com/Expensify/App/issues/16805)

### Tests

- [x] Verify that no errors appear in the JS console

### Offline tests

**Contributor+ test: Confirm the Android and iOS apps can be installed side by side**
- Install the production app from the Play Store or App Store
- Pull the branch
- Attempt to build the local build app as you normally would
- The dev app should side by side with no errors
- The dev app should contain a unique logo
- In the Settings > About page, the PR number and 'flavour' should be shown

**Internal tests: Confirm the Android and iOS apps can be installed side by side**
- Install the production app from the Play Store or App Store
- Install the AdHoc build
- Install the local dev build
- All apps should be installed side-by-side

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -140,19 +158,16 @@ android {
...
-            matchingFallbacks = ['release']
-            signingConfig signingConfigs.debug
+    }
+
+    // since we don't need variants adhocDebug and e2eDebug, we can force grade to ignore them
+    variantFilter { variant ->
+        if (variant.name == "adhocDebug" || variant.name == "e2eDebug") {
+            setIgnore(true)
         }
     }
```
Comment (Julesssss) [inline android/app/build.gradle:172]: Nice, I didn't know we could do this.

File: android/app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -66,13 +66,13 @@
...
     <style name="Theme.SplashScreen" parent="Theme.SplashScreen.Common">
-        <item name="windowSplashScreenAnimatedIcon">@android:drawable/sym_def_app_icon</item>
+        <item name="windowSplashScreenAnimatedIcon">@drawable/bootsplash</item>
         <item name="windowSplashScreenBackground">?android:colorBackground</item>
     </style>
 
     <style name="BootTheme" parent="Theme.SplashScreen">
         <item name="android:statusBarColor">@android:color/transparent</item>
-        <item name="windowSplashScreenAnimatedIcon">@mipmap/bootsplash_logo</item>
+        <item name="windowSplashScreenAnimatedIcon">@drawable/bootsplash</item>
```
Comment (Julesssss) [inline android/app/src/main/res/values/styles.xml:75]: I assume this is because the drawable icon scales in density, right?

**Review by Julesssss — COMMENTED** [review]
Looking good, I'll [compile](https://github.com/Expensify/App/actions/runs/5376287801) some builds for WIP testing. 

This icon (assets/images/new-expensify-adhoc.svg) seems to have been converted from light to dark, is that expected?

File: android/app/src/main/res/values/styles.xml (hunk 2 above)
Comment (kowczarz) [reply android/app/src/main/res/values/styles.xml:75]: Yep and we can get rid of multiple bitmaps in favour a single vector image.

**Comment by Julesssss** [comment]
Yep, in progress [here](https://github.com/Expensify/App/actions/runs/5378798841)

**Comment by Julesssss** [comment]
> ERROR:/home/runner/work/App/App/android/app/src/adhoc/res/drawable/bootsplash.xml:8: AAPT: error: resource drawable/ic_bootsplash_adhoc (aka com.expensify.chat.adhoc:drawable/ic_bootsplash_adhoc) not found

**Comment by kowczarz** [comment]
I've moved files with `ic_bootsplash`,  so android should work now, but need some more time to figure out, why iOS fails.

**Comment by Julesssss** [comment]
Yeah I'm afraid the logs are a lot more vague for iOS: 

```
Maybe the error shown is caused by using the wrong version of Xcode
Found multiple versions of Xcode in '/Applications/'
Make sure you selected the right version for your project
```

**Comment by kowczarz** [comment]
I've updatef iOS fastlane as well. @Julesssss could you please re-run the build?

**Comment by Julesssss** [comment]
Kicked off [here](https://github.com/Expensify/App/actions/runs/5390346406)

**Comment by Julesssss** [comment]
iOS: 

```
codebuild[48636:165801] Requested but did not find extension point with identifier Xcode.InterfaceBuilderBuildSupport.PlatformDefinition
```
Screenshot 2023-06-27 at 15 43 52

---------

**Android**
Screenshot 2023-06-27 at 15 42 18

**Comment by kowczarz** [comment]
@Julesssss do you have app store connect permission, so you can check if there are proper provisioning profiles and entitlements for new bundle IDs (`com.chat.expensify.chat.adhoc` and `com.chat.expensify.chat.dev`)?

**Comment by kowczarz** [comment]
@Julesssss could you please run the build - Android now should be fixed.

**Comment by Julesssss** [comment]
Hey I was off yesterday, I'll rerun the build once conflicts are resolved

**Comment by kowczarz** [comment]
Done. Any updates on AppStore connect certificates?

**Comment by Julesssss** [comment]
@kowczarz I see a few which match Adhoc and developer, I'll just confirm I'm looking at the correct ones. Also, what do you need me to check for entitlements? I can view the 'Enabled Capabilities' in the App Store, or do you mean something else?

**Comment by Julesssss** [comment]
Hmm, I was trying to get a C+ assigned.

**Comment by Julesssss** [comment]
**Android:** 

Screenshot 2023-06-29 at 14 50 58


**iOS:** 

Screenshot 2023-06-29 at 14 51 14

**Comment by kowczarz** [comment]
@Julesssss Android issue is fixed and I hope we won't face another one. For iOS I only changed adhoc action, if it will work I will need to apply the change to other actions as well.

**Comment by kowczarz** [comment]
@Julesssss Oh, I changed action code, so you need to copy my branch to internal repo, and then run the action from that branch, like @staszekscp suggested here https://github.com/Expensify/App/pull/16735#issuecomment-1490449446

**Comment by Julesssss** [comment]
@kowczarz got it, thanks. Kicked off a build using [this branch](https://github.com/Expensify/App/commits/jules-copyFlavourSupportCI)

**Comment by Julesssss** [comment]
Shared the build errors in [Slack](https://expensify.slack.com/archives/C04878MDF34/p1688135629580629)

**Comment by Julesssss** [comment]
Kicked off a new [build](https://github.com/Expensify/App/actions/runs/5624214621) 🤞

**Comment by Julesssss** [comment]
^ ignore the above I forgot to run the build from a cloned internal branch. Retrying [here](https://github.com/Expensify/App/actions/runs/5643433812)

**Comment by Julesssss** [comment]
Excellent. Noting a few small icon issue: 
- A) The App launcher icon looks odd, as if we are applying a 2nd background
- B) We should update the icon or add an indicator to the 'About' page 

A) EC0D9EA3-397E-4FA1-BB8A-FCA0ADA47692
B) 

496623D8-57D8-4B16-AB47-17E844D080CF

**Comment by Julesssss** [comment]
Thanks @kowczarz! Looks good, but it would be ideal if we could keep the PR num in addition. Something like this? 

`v1.3.45-3 PR:21553 Ad-Hoc`

**Comment by Julesssss** [comment]
I updated my cloned branch and kicked off a [new build](https://github.com/Expensify/App/actions/runs/5659104791)

**Review by Julesssss — CHANGES_REQUESTED** [review]
Looking good on both platforms! Can we just re-import `ic_launcher` and any other [icons](https://github.com/Expensify/App/blob/5e19ef7f61feaca366b55e3cb52a780d6b3b5913/android/app/src/adhoc/res/mipmap-hdpi/ic_launcher.png) that include the double border?

**Comment by kowczarz** [comment]
@Julesssss I've just fixed it.

**Review by Julesssss — CHANGES_REQUESTED** [review]
There are a few `ic_launcher_round` icons with the border still ([example](https://github.com/Expensify/App/blob/b4dd6fe7288fd56e55b083f6bdc9431f74e0aea1/android/app/src/adhoc/res/mipmap-xxhdpi/ic_launcher_round.png))

**Comment by Julesssss** [comment]
[Rebuilding...](https://github.com/Expensify/App/actions/runs/5669779331)

**Comment by Julesssss** [comment]
Perhaps this can be done separately to this PR, but it would be great if we could simultaneously install desktop apps.

**Comment by Julesssss** [comment]
Tomorrow I'm going to try the dev builds.

**Comment by kowczarz** [comment]
I think since this PR is tremendous and first commits are over 3 months old, let's consider doing this in a separate PR.

**Comment by mczernek** [comment]
> I think since this PR is tremendous and first commits are over 3 months old, let's consider doing this in a separate PR.

I'm with @kowczarz here. Especially since this might not really be needed - I am not an expert in macOS nor electron apps, but my gut tells me that since I can easily have two instances of Xcode next to each other, I should be able easily to do that with Expensify as well?

**Comment by Julesssss** [comment]
No problem, I'll create a separate issue. 

And yeah, I think it is as simple as renaming the app so that it isn't conflicting with 'New Expensify' under applications.

**Comment by Julesssss** [comment]
FYI I created the iOS dev identifier and just merged the encrypted profile. The name is: `com.expensify.chat.dev`

File: android/app/build.gradle [hunk 3]
```diff
@@ -110,6 +114,25 @@ android {
...
 
+    flavorDimensions "default"
+    productFlavors {
+        production {
+        }
+        e2e {
+            // If  are building a version that won't be uploaded to the play store, we don't have to use production keys
+            // applies all non-production flavors
+            signingConfig signingConfigs.debug
+        }
```
Comment (Julesssss) [inline android/app/build.gradle:125]: Can we apply the `.adhoc` suffix here too please? This way we can separate dev crashes from E2E in Firebase.

**Review by Julesssss — DISMISSED** [review]
Requested a few changes. Unfortunately, the Android launch icon isn't quite centered correctly, but I'm happy to perform the Android icon import myself once we've merged the PR

**Comment by Julesssss** [comment]
^ just for reference, here's what I mean about the launcher icons being slightly unequal. Again, happy to fi this myself once the main PR is merged though:

Screenshot_20230727_155159

File: android/app/src/adhoc/res/drawable/bootsplash.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android" android:opacity="opaque">
+    <item android:gravity="fill">
+        <color android:color="?attr/windowSplashScreenBackground" />
+    </item>
+
+    <item android:gravity="center" android:drawable="@drawable/ic_bootsplash_adhoc"/>
```
[the remark this answers is not in the document]
Comment (kowczarz) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: I think I know what cause this issue, but I need to test it on different simulators with different dpi, will check it tomorrow morning.

**Comment by Julesssss** [comment]
> To generate icons I was using [icon.kitchen](https://icon.kitchen) and the problem is that width and height of the icons with badges are slightly different so I had to specify the padding by eye.

No worries, I'll use the Android Studio import tool after we've merged this PR

**Review by Julesssss — COMMENTED** [review]
Created a small [follow-up PR](https://github.com/Expensify/App/pull/23923/files) to improve the app icon padding. The one remaining task before the final review is to resolve the splash screen issue 🎉

File: android/app/src/adhoc/res/drawable/bootsplash.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Julesssss) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: Yeah, I'm seeing this on a Pixel 6A. I can't reproduce on older devices like a Nexus 5X.

[the remark this answers is not in the document]
Comment (kowczarz) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: @Julesssss is that a physical device, or emulator?

[the remark this answers is not in the document]
Comment (Julesssss) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: Physical

[the remark this answers is not in the document]
Comment (Julesssss) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: I have a spare couple of hours today so I'll help try to resolve this.

**Review by Julesssss — COMMENTED** [review]
[This fix](https://github.com/Expensify/App/pull/21551#discussion_r1280418803) seems to work, but I'll confirm on a physical device after lunch (edit -- it doesn't 😭 )

File: android/app/src/adhoc/res/drawable/bootsplash.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kowczarz) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: It seems this crashes the app

[the remark this answers is not in the document]
Comment (Julesssss) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: Pretty sure the xml is correct, I assume the crash is because the assets need to be renamed. Let me push a commit to a branch after lunch

[the remark this answers is not in the document]
Comment (kowczarz) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: In the meantime I've fixed the logo flashing from 0:02-0:03

[the remark this answers is not in the document]
Comment (kowczarz) [reply android/app/src/adhoc/res/drawable/bootsplash.xml:8]: Yes, that sounds good. I will revert the changes tomorrow morning.

File: src/components/SplashScreenHider/index.native.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ const defaultProps = {
     onHide: () => {},
 };
 
+
```
Comment (Julesssss) [inline src/components/SplashScreenHider/index.native.js:18]: I think this is triggering the lint error: 

```
src/components/SplashScreenHider/index.native.js
Error: Prettier diff detected! Please run `npm run prettier` and commit the changes.
```

Comment (Julesssss) [reply src/components/SplashScreenHider/index.native.js:18]: Just an unnecessary new line

**Comment by Julesssss** [comment]
Kicked off a new build for final testing

File: android/app/src/main/res/drawable/ic_bootsplash.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="100dp"
+    android:height="100dp"
+    android:viewportWidth="100"
+    android:viewportHeight="100">
+  <path
+      android:pathData="M74,15.63a42.03,42.03 0,0 0,-24 -7.5A41.77,41.77 0,0 0,8.12 49.87c0,9 2.76,17.26 7.5,24a4.35,4.35 0,0 1,-0.37 5.63c-1.75,1.63 -4.5,1.5 -5.88,-0.38A49.99,49.99 0,0 1,50 0c10.88,0.01 20.88,3.51 29.25,9.38 2,1.5 2,4.26 0.38,5.88a4.35,4.35 0,0 1,-5.63 0.38ZM67.62,24.87L67.62,35.5L45.25,35.5v9.13h18v10.62h-18v9.63L67.5,64.88v10.37L32.25,75.25L32.25,24.87h35.38ZM91.12,21.75a3.78,3.78 0,0 0,-5.87 -0.38,4.35 4.35,0 0,0 -0.5,5.5c4.38,6.63 7,14.63 7,23.13 0,8.5 -2.63,16.75 -7.25,23.5 -1.13,1.75 -1.13,4.13 0.38,5.63 1.75,1.75 4.5,1.62 5.87,-0.38A49.5,49.5 0,0 0,99.88 50a47.66,47.66 0,0 0,-8.75 -28.25ZM26.88,84.88c6.63,4.37 14.63,7 23.13,7 8.63,0 16.75,-2.75 23.5,-7.25a4.35,4.35 0,0 1,5.63 0.37c1.75,1.75 1.62,4.5 -0.38,5.88A50.37,50.37 0,0 1,50 100a49.75,49.75 0,0 1,-28.25 -8.75c-2,-1.38 -2.13,-4.13 -0.38,-5.88 1.38,-1.5 3.75,-1.62 5.5,-0.5Z"
+      android:fillColor="#002E22"
+      android:fillType="evenOdd"/>
+</vector>
```
Comment (Julesssss) [inline android/app/src/main/res/drawable/ic_bootsplash.xml:10]: Is this file still necessary? I think it might not be used since we reverted the splash changes.

File: android/app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -66,7 +66,7 @@
     </style>
 
     <style name="Theme.SplashScreen" parent="Theme.SplashScreen.Common">
-        <item name="windowSplashScreenAnimatedIcon">@android:drawable/sym_def_app_icon</item>
+        <item name="windowSplashScreenAnimatedIcon">@drawable/bootsplash</item>
```
Comment (Julesssss) [inline android/app/src/main/res/values/styles.xml:69]: Also curious about this change, is this still necessary?

Comment (kowczarz) [reply android/app/src/main/res/values/styles.xml:69]: I didn't spot it.

File: android/app/src/main/res/drawable/ic_bootsplash.xml (hunk 6 above)
Comment (kowczarz) [reply android/app/src/main/res/drawable/ic_bootsplash.xml:10]: Nope, missed that file, I'm going through all files and checking if I didn't miss anything else.

**Comment by Julesssss** [comment]
Hey, just to let you know I'll be away until Monday. I'm hoping to run the final tests and review on Monday, then after merging will quickly follow up with the icon PR update. 

This also means that I will be online when the changes go live and I can explain the changes to everyone 👍

File: android/app/build.gradle [hunk 8]
```diff
@@ -110,6 +114,26 @@ android {
         versionName "1.3.46-1"
     }
 
+    flavorDimensions "default"
+    productFlavors {
+        production {
+        }
```
Comment (Julesssss) [inline android/app/build.gradle:120]: Are we just leaving this empty object to make it clearer to other devs that it COULD be used? if so can you please just add a small comment inside to explain? Thanks!

File: ios/Podfile.lock [hunk 9]
```diff
@@ -1141,6 +1141,6 @@ SPEC CHECKSUMS:
   Yoga: f87bd008d34c18bf990d4371200802f2866ce18c
   YogaKit: f782866e155069a2cca2517aafea43200b01fd5a
 
-PODFILE CHECKSUM: 69f576e475be6d130ee96103b131f71bfdbf8a6e
+PODFILE CHECKSUM: 054ef63292919db072e55b622ee63b87c83d39a0
 
-COCOAPODS: 1.11.3
+COCOAPODS: 1.12.1
```
Comment (Julesssss) [inline ios/Podfile.lock:1146]: Was this expected? I'm not sure we should bump cocoapods within this PR unless absolutely necessary.

Comment (kowczarz) [reply ios/Podfile.lock]: AFAIK On `main` we are using `1.12.1` https://github.com/Expensify/App/blob/main/ios/Podfile.lock#L1237

File: android/app/build.gradle (hunk 8 above)
Comment (kowczarz) [reply android/app/build.gradle]: I did some cleanup here.

**Review by allroundexperts — APPROVED** [review]

**Review by Julesssss — COMMENTED** [review]
Thanks for your patience @kowczarz. Just two more things to check off before we can merge: 
- [Question](https://expensify.slack.com/archives/C04878MDF34/p1691592215605039) about the type of iOS profile I used for AdHoc builds
- [Android only] The Main page icon shows Dev for the AdHoc builds: 

|  Main page  | About page |
| ----------- | ----------- |
| Screenshot_20230809-160019 | Screenshot_20230809-160025 |

**Comment by Julesssss** [comment]
@allroundexperts could you please complete the checklist and add screenshots for Android? I'll handle iOS

**Review by allroundexperts — APPROVED** [review]

**Review by Julesssss — DISMISSED** [review]
Unfortunately, the latest commit didn't fix the Ad Hoc build showing the dev logo. This is the last blocker though! 

Screenshot_20230810-152025

**Review by Julesssss — APPROVED** [review]
Final logo issue was resolved 👍 

Screenshot_20230811_150605

**Comment by Julesssss** [comment]
Okay lets merge! Now is a good time as we likely won't deploy today and I have time to update the icons. I'll continue to work on the additional tasks in the meantime.

**Review by situchan — COMMENTED** [review]
Is it intentional to add `main.jsbundle.map` file to git?

**Comment by Julesssss** [comment]
Nope, thanks. I'll clean it up.

File: android/app/build.gradle [hunk 10]
```diff
@@ -111,20 +136,17 @@ android {
             signingConfig signingConfigs.debug
         }
         release {
-            signingConfig signingConfigs.release
+            signingConfig signingConfigs.debug
+            productFlavors.production.signingConfig signingConfigs.release
```
Comment (Julesssss) [inline android/app/build.gradle:140]: Hey @kowczarz, do you recall why line 140 is necessary? Is this doing something that the following code would not? 

```
signingConfig signingConfigs.release
```

**Comment by mvtglobally** [comment]
@Julesssss are we good to check this off?

**Comment by WoLewicki** [comment]
Hey, coming from the new arch intergration PR (https://github.com/Expensify/App/pull/13767), we need to get some changes in order for the app to build and work correctly. All of them should be contained in this commit: https://github.com/Expensify/App/pull/13767/commits/3b6a38775838c580f8e0e1f134f68e526502fc35. cc @kowczarz and @Julesssss for visibility.

They are needed since, for some reason, on new arch, the spaces in the scheme names are not accepted by one of the scripts. Also, there was a minor issue with `Android` build.

**Comment by Julesssss** [comment]
Hi @WoLewicki, I just got back from holiday. Could you please raise a PR for that? Kamil mentioned something similar to me and I created an issue [here](https://github.com/Expensify/App/issues/26152) if that helps.

**Comment by mananjadhav** [comment]
We changed our dev logo in this PR `assets/images/new-expensify-dev.svg`, which caused a regression [here](https://github.com/Expensify/App/issues/25617) (Dev logo is grayed out). I couldn't comment on the file change, hence I am commenting here.
